### DISCUSSION TITLE
How do you correct the image `source` in a `tileset` with the excalibur-tiled plugin?

### DISCUSSION THREAD

**Discussion by AWaselnuk** [excaliburjs/Excalibur#2561] [Q&A] [answered]
Hey everyone! I'm new to game dev and Excalibur. Resolving most issues so far thanks to the clear docs. Thank you!

I'm stuck on my workflow with Tiled right now. When I export my `.tmx` file the references to tilesheets are not correct. This is a known issue, easily resolved with `convertPath` as described [here](https://github.com/excaliburjs/excalibur-tiled#handling-tiled-paths).

My problem is that, this only fixes the source references for the Tilemap, but not the tilesheets. Now the tilesheets have broken references to their source images, and I can't figure out how to rewrite the paths in those `.tsx` files.

Example of my `.tsx` file with the `source` attribute I need to rewrite:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<tileset version="1.9" tiledversion="1.9.2" name="kenney-roguelike-pack-16px" tilewidth="16" tileheight="16" spacing="1" tilecount="1767" columns="57">
 <image source="/THIS/IS/NOT/CORRECT.png" width="968" height="526"/>
</tileset>
```

I might be confused about how to export tilesheets from Tiled or I might be missing some other `convertPath` method. Any guidance would be much appreciated.

**Comment by AWaselnuk** [comment, marked as answer]
Hmmm with further `console.log` slinging it looks like the `convertPath` method actually is getting called on the `.tsx` files as well. I had some bad logic that couldn't handle the different `basePath` correctly.

My mistake!

**Reply by eonarheim** [reply]
@AWaselnuk I agree that this is a little tricky, we'll revisit this to see if there is a way to make this more bullet proof without need to mess with `convertPath`
